EIPS/eip-6046.md

```diff
@@ -0,0 +1,63 @@
+---
+eip: 6046
+title: Replace SELFDESTRUCT with DEACTIVATE
+description: Change SELFDESTRUCT to not delete storage keys and use a special value in the account nonce to signal deactivation
+author: Alex Beregszaszi (@axic)
+discussions-to: https://ethereum-magicians.org/t/almost-self-destructing-selfdestruct-deactivate/11886
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-11-25
+requires: 2681
+---
+
+## Abstract
+
+Change `SELFDESTRUCT` to not delete all storage keys, and to use a special value in the account nonce to signal *decativated* accounts. Because the semantics of revival change (storage keys may exists), we also rename the instruction to `DEACTIVATE`.
+
+## Motivation
+
+The `SELFDESTRUCT` instruction currently has a fixed price, but is unbounded in terms of how many storage/account changes it performs (it needs to delete all keys). This is an outstanding concern for long.
+
+Furthermore with *Verkle trees* accounts will be organised differently: account properties, including storage, will have individual keys, and it will not be possible to traverse and find all used keys. This makes `SELFDESTRUCT` very challenging to support in Verkle trees.
+
+## Specification
+
```

Comment:
This should define the block number on which the upgrade occurs.

Reply:
This practice has long been gone, because block numbers are decided outside of the specifications. Furthermore, upgrades may not even be rolled out at pre-defined block numbers anymore.

Reply:
Not sure what the policy is on this anymore. I'll resolve this.